### PR TITLE
Add a prefix for prometheus (needed by CMS)

### PR DESCRIPTION
**Pull request by ericvaandering**
This is what remains of PR50 after fixing the immediate issues with the probe

### CODE REVIEW

**Comment by ericvaandering** [comment]
@dchristidis can you take a look at this? We have other things reliant on this. It should be a no-op if these settings are not set.

**Comment by dchristidis** [comment]
We need a solution for where to place commonly-used code in a way that’s available across all probes (not merely `common`).

**Comment by ericvaandering** [comment]
Ok, where do you suggest? A utils directory that is added to the path?

**Comment by dchristidis** [comment]
Yes, something like `sys.path.append('../utils')` is the only suggestion I found, but didn’t spend enough time on time. I am very open to suggestions.

**Comment by ericvaandering** [comment]
OK, how about this. Any VO which wants to use this functionality would have to include the utils.py in their directory as well and import it.

Not part of this PR, but I had a thought that a better way to do all this Prometheus boiler plate (including what this PR adds) is to add a prometheus push gateway context manager which will push everything at the end.

**Comment by ericvaandering** [comment]
@dchristidis ^^^

**Comment by dchristidis** [comment]
Thank you for your patience. I’ve had the time to think about this and I like it.

I was thinking of some small changes which don’t necessarily make it better, so I wanted to run them by you:
 1. Inside `common`, we create `utils/{__init__.py,common.py}`.
 2. Inside community-specific areas (e.g. `cms`), we create `utils.py` (as you suggested) with the (adjusted) path manipulation in your PR and `import common`.

Effectively, it’s a rename of `probe_utils` to `common` and a change of location (inside `common` as opposed to outside it). My main motivation is the contributor’s guide and the significance of `common` when it comes to the procedures.

What do you think?

**Comment by ericvaandering** [comment]
I like this. And it keeps the common probes looking "nice" with their imports. Shall I make the change?

**Comment by dchristidis** [comment]
Please do. In addition, I would suggest:
 1. `python3` in the hashbangs.
 2. Type hints for the newly-defined function.

**Comment by ericvaandering** [comment]
Isn't python =>python2 in cc7 still?

**Comment by dchristidis** [comment]
May I ask for your motivation? Having `python` instead of `python3` is 1) inconsistent with the Rucio repository, 2) problematic for the current RHEL7-based ATLAS setup where `python` refers to `python2` 3) against PEP 394 (I think).

**Comment by ericvaandering** [comment]
The second commit is what this would look like with a context manager to hide almost all of the prometheus boiler-plate. What do you think?

**Comment by ericvaandering** [comment]
OK, here are the type hints too.

File: common/check_expired_rules [hunk 1]
```diff
@@ -17,12 +17,12 @@ Probe to check the backlog of expired rules.
...
 import traceback
 
-from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
-
-from rucio.common.config import config_get
+from prometheus_client import CollectorRegistry, Gauge
 from rucio.core import monitor
 from rucio.db.sqla.session import BASE, get_session
 
+from utils.common import PrometheusPusher
```
[the remark this answers is not in the document]
Comment (ericvaandering) [reply common/check_expired_rules:24]: Will check and fix.

File: common/utils/common.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import json
+import sys
+from typing import Tuple, Dict, List, Optional
+
+from prometheus_client import push_to_gateway
+from rucio.common.config import config_get
+
+
+def get_prometheus_config() -> Tuple[List, str, Dict]:
```
[the remark this answers is not in the document]
Comment (dchristidis) [reply common/utils/common.py:19]: Yes, that’s perfectly fine. Thanks!

**Comment by ericvaandering** [comment]
OK, fixed the init.py and added a pattern verified to work for the import in both common and cms directories.

**Review by dchristidis — APPROVED** [review]

**Pull request merged**